pkg/release/release.go

```diff
@@ -128,78 +128,105 @@ func (r *Release) Condition() ConditionSet {
 func (r *Release) Delete(ctx context.Context, name string) error {
 	releaseName := fmt.Sprintf("%s-%s", r.namespace, name)
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleting release %#q", releaseName))
-
 	err := r.helmClient.DeleteRelease(releaseName, helm.DeletePurge(true))
 	if helmclient.IsReleaseNotFound(err) {
-		return microerror.Maskf(releaseNotFoundError, releaseName)
+		return microerror.Maskf(releaseNotFoundError, "failed to delete release %#q", name)
 	} else if helmclient.IsTillerNotFound(err) {
 		return microerror.Mask(tillerNotFoundError)
 	} else if err != nil {
 		return microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleted release %#q", releaseName))
-
 	return nil
 }
 
-func (r *Release) EnsureDeleted(ctx context.Context, name string) error {
-	releaseName := fmt.Sprintf("%s-%s", r.namespace, name)
+// EnsureDeleted makes sure the release is deleted and purged and all
+// conditions are met.
+func (r *Release) EnsureDeleted(ctx context.Context, name string, conditions ...func() error) error {
```

Comment:
`EnsureDeleted` and `EnsureInstalled` are the new functions here. Diff is ugly, maybe look at the whole file.